libs/remix-tests/src/runTestSources.ts

```diff
@@ -5,133 +5,142 @@ import { deployAll } from './deployer'
 import { runTest } from './testRunner'
 
 import Web3 from 'web3'
+import EventManager from './lib/eventManager'
 import { Provider, extend } from '@remix-project/remix-simulator'
 import {
   FinalResult, SrcIfc, compilationInterface, ASTInterface, Options,
   TestResultInterface, AstNode, CompilerConfiguration
 } from './types'
 require('colors')
 
-const createWeb3Provider = async function () {
-  const web3 = new Web3()
-  const provider: any = new Provider()
-  await provider.init()
-  web3.setProvider(provider)
-  extend(web3)
-  return web3
-}
+export class UnitTestRunner {
```

Comment:
any chance we can discard `runTestFile` and use this instead?

Reply:
That is there for CLI usage. I would like to do that in a separate PR. Will be tracked by https://github.com/ethereum/remix-project/issues/1561